consul/consul.go

```diff
@@ -33,7 +34,11 @@ func getServiceFromConsul(getService func(string, string, bool, *api.QueryOption
 
 func extractNodesAddresses(entries []*api.ServiceEntry) (nodesAddr []string) {
 	for i := range entries {
-		nodesAddr = append(nodesAddr, fmt.Sprintf("%s:%d", entries[i].Service.Address, entries[i].Service.Port))
+		addr := entries[i].Service.Address
+		if addr == "" {
```

Comment:
document under which conditions service address can be empty here

Reply:
I don't know under which conditions it can happen but it seems like a normal behavior of Consul: https://www.consul.io/api-docs/agent/service#sample-response

Reply:
I'm just wondering why you changed the code here. Is it because you have been facing empty service address ? If yes I would be curious to know the underlying reason.